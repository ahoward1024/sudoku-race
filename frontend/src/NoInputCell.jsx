import React, {Component} from 'react';
import PropTypes from 'prop-types';

class NoInputCell extends Component {
  constructor(props) {
    super(props);
    this.state = {
      'id': this.props.id,
      'value': this.props.value
    };
  }

  render() {
    return (
      <input
        key={this.state.id}
        id={this.state.id}
        className="cell-input noinput"
        maxLength="1"
        value={this.state.value}
        readOnly="true"
      />
    );
  }
}

NoInputCell.propTypes = {
  'id': PropTypes.string,
  'value': PropTypes.string
};

export default NoInputCell;
