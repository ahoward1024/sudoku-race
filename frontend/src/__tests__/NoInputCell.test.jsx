import React from 'react';
import renderer from 'react-test-renderer';
import NoInputCell from '../NoInputCell';

test('Create a NoInputCell', () => {
  const tree = renderer.create(<NoInputCell value="1"/>).toJSON();
  expect(tree).toMatchSnapshot();
});
