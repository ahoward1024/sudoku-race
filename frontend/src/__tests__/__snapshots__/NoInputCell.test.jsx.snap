// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a NoInputCell 1`] = `
<input
  className="cell-input noinput"
  id={undefined}
  maxLength="1"
  readOnly="true"
  value="1"
/>
`;
